### PR TITLE
[flutter_tools] add tool feature/scaffold for UWP

### PR DESCRIPTION
**Pull request by jonahwilliams**
I named this Windows UWP (and similar), please LMK if that is the appropriate name. This does not include any doctor or flutter build scaffolding for now. 

https://github.com/flutter/flutter/issues/70211
https://github.com/flutter/flutter/issues/14967

### CODE REVIEW

File: packages/flutter_tools/lib/src/windows/windows_device.dart [hunk 1]
```diff
@@ -68,25 +70,81 @@ class WindowsDevice extends DesktopDevice {
   }
 }
 
+// A device that represents a desktop Windows target.
```
Comment (cbracken) [inline packages/flutter_tools/lib/src/windows/windows_device.dart:73]: a desktop Windows UWP target.

File: packages/flutter_tools/lib/src/features.dart [hunk 2]
```diff
@@ -54,6 +54,9 @@ abstract class FeatureFlags {
   /// Whether the CFE experimental invalidation strategy is enabled.
   bool get isExperimentalInvalidationStrategyEnabled => true;
 
+  /// Whether the windows UWP embedding is enabled.
+  bool get isWindowsUWPEnabled => false;
```
Comment (cbracken) [inline packages/flutter_tools/lib/src/features.dart:58]: Very nitpicky and I bet we're inconsistent about this, and also have lots of exceptions to Effective Dart in our code, but... generally use CamelCase for abbreviations/initialisms longer than two characters. e.g. `Url`, `HttpClient`.


https://dart.dev/guides/language/effective-dart/style#do-capitalize-acronyms-and-abbreviations-longer-than-two-letters-like-words

File: packages/flutter_tools/test/general.shard/features_test.dart [hunk 3]
```diff
@@ -448,6 +448,34 @@ void main() {
 
       expect(featureFlags.isWindowsEnabled, true);
     });
+
+    // Windows UWP desktop
+
+    testWithoutContext('flutter windows UWP desktop off by default on master', () {
```
Comment (cbracken) [inline packages/flutter_tools/test/general.shard/features_test.dart:454]: nit^3: Windows :P

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

File: packages/flutter_tools/test/general.shard/features_test.dart (hunk 3 above)
Comment (cbracken) [reply packages/flutter_tools/test/general.shard/features_test.dart:454]: I mean technically Flutter too, I suppose. Now I feel conflicted.

**Comment by jonahwilliams** [comment]
Thanks @cbracken ! I'll leave this open for now to give @clarkezone a chance to take a look

File: packages/flutter_tools/test/general.shard/features_test.dart (hunk 3 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/features_test.dart]: Fixed!

File: packages/flutter_tools/lib/src/features.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/features.dart]: Fixed. In my defense, this one is hard to follow :)

File: packages/flutter_tools/lib/src/windows/windows_device.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/windows/windows_device.dart]: Done

**Review by clarkezone — APPROVED** [review]
LGTM
